src/firefox/remote.js

```diff
@@ -43,133 +41,138 @@ export type FirefoxRDPResponseAny = Object;
+    }
+
+    try {
+      const response = await this.client.request('listTabs');
+      // addonsActor was added to listTabs in Firefox 49 (bug 1273183).
+      if (response.addonsActor == null) {
+        log.debug(
+          'listTabs returned a falsey addonsActor: ' +
+          `${JSON.stringify(response)}`);
+        return Promise.reject(new RemoteTempInstallNotSupported(
+          'This is an older version of Firefox that does not provide an ' +
+          'add-ons actor for remote installation. Try Firefox 49 or ' +
+          'higher.'));
+      }
+      return response.addonsActor;
+    } catch (err) {
+      log.debug('listTabs error', err);
+      const message = requestErrorToMessage(err);
+      throw new WebExtError(`Remote Firefox: listTabs() error: ${message}`);
+    }
   }
 
-  installTemporaryAddon(
+  async installTemporaryAddon(
     addonPath: string
   ): Promise<FirefoxRDPResponseAddon> {
-    return new Promise((resolve, reject) => {
-      this.getAddonsActor().then((addonsActor) => {
-        this.client.client.makeRequest({
-          to: addonsActor,
-          type: 'installTemporaryAddon',
-          addonPath,
-        }, (installResponse) => {
-          if (installResponse.error) {
-            return reject(new WebExtError(
-              'installTemporaryAddon: Error: ' +
-              `${installResponse.error}: ${installResponse.message}`));
-          }
-          log.debug(
-            `installTemporaryAddon: ${JSON.stringify(installResponse)}`);
-          log.info(`Installed ${addonPath} as a temporary add-on`);
-          resolve(installResponse);
-        });
-      }).catch(reject);
-    });
+    const addonsActor = await this.getAddonsActor();
+
+    try {
+      const response = await this.client.request({
+        to: addonsActor,
+        type: 'installTemporaryAddon',
+        addonPath,
+      });
+      log.debug(`installTemporaryAddon: ${JSON.stringify(response)}`);
+      log.info(`Installed ${addonPath} as a temporary add-on`);
+      return response;
+    } catch (err) {
+      const message = requestErrorToMessage(err);
+      throw new WebExtError(`installTemporaryAddon: Error: ${message}`);
+    }
   }
 
-  getInstalledAddon(addonId: string): Promise<FirefoxRDPAddonActor> {
-    return new Promise(
-      (resolve, reject) => {
-        this.client.request('listAddons', (error, response) => {
-          if (error) {
-            reject(new WebExtError(
-              `Remote Firefox: listAddons() error: ${error}`));
-          } else {
-            resolve(response.addons);
-          }
-        });
-      })
-      .then((addons) => {
-        for (const addon of addons) {
-          if (addon.id === addonId) {
-            return addon;
-          }
+  async getInstalledAddon(addonId: string): Promise<FirefoxRDPAddonActor> {
+    try {
+      const response = await this.client.request('listAddons');
+      if (response.addons == null) {
```

Comment:
Why this explicit check? This code shouldn't ever be reached, the backend should send the `noAddons` error if there are no add-ons, and otherwise the error becomes `Remote Firefox: listAddons() error: addons is not iterable`, which is sufficiently clear.

Reply:
When I did read this comment I was convinced that I may have added the explicit check on `response.addons == null` because of flowtype errors, but flow didn't seem to care at the moment, and so I'm removing it (along with the related unit test)